web/ui/react-app/src/pages/targets/JobDetails.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FC } from 'react';
+import { TargetGroup } from './target';
+import { Table } from 'reactstrap';
+import StateIndicator from './StateIndicator';
+import EndpointLink from './EndpointLink';
+import { humanizeDuration, formatRelative, now } from '../../utils/timeFormat';
+import SeriesLabels from './SeriesLabels';
+import styles from './JobDetails.module.css';
+
+export const columns = ['Endpoint', 'State', 'Labels', 'Last Scrape', 'Scrape Duration', 'Error'];
+
+interface JobDetailsProps {
+  targetGroup: TargetGroup;
+}
+
+const JobDetails: FC<JobDetailsProps> = ({ targetGroup }) => {
+  return (
+    <Table className={styles.table} size="sm" bordered hover striped>
+      <thead>
+        <tr key="header">
+          {columns.map(column => (
+            <th key={column}>{column}</th>
+          ))}
+        </tr>
+      </thead>
+      <tbody>
+        {targetGroup.targets.map(target => {
+          const { discoveredLabels, labels, scrapeUrl, lastError, lastScrape, lastScrapeDuration, health } = target;
+
+          return (
+            <tr key={scrapeUrl}>
+              <td className={styles.endpoint}>
+                <EndpointLink endpoint={scrapeUrl} />
+              </td>
+              <td className={styles.state}>
+                <StateIndicator health={health} />
+              </td>
+              <td className={styles.labels}>
+                <SeriesLabels discoveredLabels={discoveredLabels} labels={labels} />
+              </td>
+              <td className={styles['last-scrape']}>{formatRelative(lastScrape, now())}</td>
```

Comment:
`now()` is cool for now (pun intended), though it's possible that the user's clock is totally off from the server's clock and these durations will be totally wrong. Long run we could maybe get the server's reference "now" timestamp via the `Date` header that Prometheus returns on each API call (just a wild idea without having to add more infos to APIs). But forget that for now :)